### DISCUSSION TITLE
Partitioning of stress due to gravity wave drag and mountain blocking.

### DISCUSSION THREAD

**Discussion by bnepal22** [ufs-community/ufs-weather-model#2289] [Q&A] [answered]
I am trying to see the orographic gravity wave drag impact in UFS model by tuning the cdmbgwd parameter. However I am not quiet sure how to partition the mountain blocking and gravity wave drag impacts. I am able to get the output 'dusfc', 'dvsfc' , 'u-gwd_ave' and 'v-gwd_ave'. Could someone guide me how to partition the output stress into due to gravity wave drag and due to mountain blocking.

**Comment by gspetro-NOAA** [comment]
Hello @bnepal22 !

I just wanted to update you that I have reached out to a subject matter expert for some assistance with your question. I'm hoping to hear back from him soon and will update you once I do. 

Best,
Gillian Petro

**Reply by bnepal22** [reply]
I  am waiting for your response.

**Reply by gspetro-NOAA** [reply]
Hi @bnepal22 , 

Sorry for the delay--I am likewise waiting on the SME's response. I will follow up and see if I can get a response, but I can't guarantee anything. Most of our SMEs do this user support work as a courtesy on the side, and during very busy development cycles, it can be difficult to hear from them because they need to prioritize development of operational weather systems. 

Best,
Gillian

**Comment by mdtoyNOAA** [comment]
@bnepal22 Sorry for the delay in my response.  To configure the model to provide diagnostic output of the orographic gravity wave drag and blocking, please do the following:
1)  In the namelist file 'input.nml', set the following 2 variables:

     ldiag3d=.true.
     ldiag_ugwp=.true.
(or if you are using the workflow, in config.fcst, use:  'export ldiag3d=.true.' and 'export ldiag_ugwp=.true.')


2) Add the following lines to the 'diag_table' file:

# Diagnostic variables for Unified Gravity Wave Physics (UGWP)
"gfs_phys",    "dws3dt_ogw",  "dws3dt_ogw",     "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "dws3dt_obl",  "dws3dt_obl",     "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "ldu3dt_ogw",  "ldu3dt_ogw",     "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "ldu3dt_obl",  "ldu3dt_obl",     "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "du3_ogwcol",  "du3_ogwcol",     "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "dv3_ogwcol",  "dv3_ogwcol",     "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "du3_oblcol",  "du3_oblcol",     "fv3_history2d",  "all",  .false.,  "none",  2
"gfs_phys",    "dv3_oblcol",  "dv3_oblcol",     "fv3_history2d",  "all",  .false.,  "none",  2

The variables 'dws3dt_*' are the time-averaged 3-dimensional fields of windspeed tendency of 'ogw' (orographic GWD) and 'obl' (blocking).  (These will make your output files pretty large.  You can omit these lines if you like.)
The variables 'ldu3dt_*' are the 3-dimensional time-averaged zonal-component of the 'ogw' and 'obl' drag tendencies.
The variables du3_ogwcol and du3_oblcol are the 2-dimensional (horizontal) time-averaged surface stresses from 'ogw' and 'obl' respectively.  The surface stresses are the vertically integrated drag tendencies in each column.

Please let me know if this works and/or if you have any questions.  Thank you.

**Reply by bnepal22** [reply]
Thank you very much, I will give try to above and will get back to you.

**Comment by bnepal22** [comment]
@mdtoyNOAA  I tried to perform the simulation with the above list added in the diag_table and updating  &gfs_physics_nml
with       ldiag3d=.true.
             ldiag_ugwp=.true.
Though I couldnt get the variable you suggested, I was able to get the output for  tau_mtb, tau_ogw and tau_tofd.
float tau_mtb(time, grid_yt, grid_xt) ;
                tau_mtb:long_name = " ORO-MTB integrated flux from surface" ;
                tau_mtb:units = "N/m2" ;
                tau_mtb:missing_value = 9.99e+20f ;
                tau_mtb:_FillValue = 9.99e+20f ;
                tau_mtb:cell_methods = "time: point" ;
        float tau_owg(time, grid_yt, grid_xt) ;
                tau_owg:long_name = " OGW vertical MF at launch level" ;
                tau_owg:units = "N/m2" ;
                tau_owg:missing_value = 9.99e+20f ;
                tau_owg:_FillValue = 9.99e+20f ;
                tau_owg:cell_methods = "time: point" ;
        float tau_tofd(time, grid_yt, grid_xt) ;
                tau_tofd:long_name = " ORO-TOFD integrated flux from surface" ;
                tau_tofd:units = "N/m2" ;
                tau_tofd:missing_value = 9.99e+20f ;
                tau_tofd:_FillValue = 9.99e+20f ;
                tau_tofd:cell_methods = "time: point" ;
I am using "FV3_GFS_v15p2" of ccpp_suite and using stretched grid.

**Reply by mdtoyNOAA** [reply, marked as answer]
@bnepal22  Oh, I didn't realize you were using an older version of the gravity wave drag parameterization.  The variables I mentioned to you are not included in the version you're using.  You would need to switch to a newer CCPP suite for the GFSv17 prototypes, such as 'FV3_GFS_v17_p8_ugwpv1'.  However, it looks like the variables you mentioned are what you needed.  I hope this will be helpful.

**Reply by bnepal22** [reply]
@mdtoyNOAA  Though I was able to get the tau_mtb, tau_ogw and tau_tofd, all values are null values. May Be I will think of updating the version. Thank you very much for your help.

**Comment by gspetro-NOAA** [comment]
@bnepal22 Do you need further assistance, or did this information answer your question?

**Reply by bnepal22** [reply]
@gspetro-NOAA,  Sorry for late response. Yes, I think above information answered my query. Thank you for your support. I will get back to you if need any further assistance.
